metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/resolver.py

```diff
@@ -352,7 +460,11 @@ def resolve_to_data_platform_table_list(self) -> List[DataPlatformTable]:
                 continue
 
             table_full_name_creator: AbstractDataPlatformTableCreator = (
-                supported_resolver.get_table_full_name_creator()()
+                supported_resolver.get_table_full_name_creator()(
+                    ctx=ctx,
```

Comment:
eventually, will it be possible to get rid of this `AbstractDataPlatformTableCreator` / move some of that logic into the sqlglot_lineage parser itself?

Overall I found it pretty tricky to follow this code and understand exactly what it was doing

Reply:
AbstractDataPlatformTableCreator is an abstraction over M-Query parsing and different classes are extending it to support special construct of M-Query, like RedShift, Snowflake, Postgres, MSSQL and some of them has native query inside M-Query code. In current code SNOWFLAKE and AMAZON_REDSHIFT supports M-Query way to create transformation as well as custom sql to create transformation inside M-Query. Both are using `AbstractDataPlatformTableCreator.parse_custom_sql` to generate lineage if during M-Query parsing they found custom query inside M-Query

Reply:
Please add some comments to this file about how it is structured at a high level and why it was done that way

Reply:
done